cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -217,22 +217,25 @@ const dialTimeout = 40 * time.Second
 
 // Sync synchronizes client's endpoints with the known endpoints from the etcd membership.
 func (c *Client) Sync() error {
-	cli, err := clientv3.New(clientv3.Config{
-		Endpoints:   c.Endpoints,
-		DialTimeout: dialTimeout,
-		DialOptions: []grpc.DialOption{
-			grpc.WithBlock(), // block until the underlying connection is up
-		},
-		TLS: c.TLS,
-	})
-	if err != nil {
-		return err
-	}
-	defer cli.Close()
-
 	// Syncs the list of endpoints
+	var cli *clientv3.Client
 	var lastError error
-	err = wait.ExponentialBackoff(etcdBackoff, func() (bool, error) {
+	err := wait.ExponentialBackoff(etcdBackoff, func() (bool, error) {
+		var err error
+		cli, err = clientv3.New(clientv3.Config{
+			Endpoints:   c.Endpoints,
+			DialTimeout: dialTimeout,
```

Comment:
for MemberAdd we made this equal to `etcdTimeout`, which means that we can potentially remove the `dialTimeout` var.

the note on https://github.com/kubernetes/kubernetes/blob/64c70a7f0f4a55398b6bc7fc1be0fde3dcaa97bb/cmd/kubeadm/app/util/etcd/etcd.go#L213-L216

is concerning, but maybe our new retries scheme would be beneficial for the arm64 issue too.

Reply:
+1  let's do it.
If there's a problem, we can revert it back.